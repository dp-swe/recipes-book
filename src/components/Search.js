import React from 'react';
import {connect} from 'react-redux';
import {getRecipes} from '../store/ducks/recipesDuck';
import styles from './styles/search.module.scss';
import {ReactComponent as IconSearch} from '../static/icons/search.svg';

export const Search = ({submit}) => {

    const [search, setSearch] = React.useState("");
    
    const handleSubmit = (e) => {
        e.preventDefault();
        if(search.length > 3){
            submit(search);
        }
    }

    return (
        <div className={styles.search}>
            <form onSubmit={handleSubmit} id='searchForm'>
                <div className={styles.search__input_parent}>
                    <IconSearch className={styles.search__icon}/>
                    <input
                        id='searchInput'
                        className={styles.search__input}
                        value={search}
                        onChange={(e) => setSearch(e.target.value)}
                    />
                </div>
            </form>
        </div>
    )
}

const mapDispatchToProps = dispatch => ({
    submit: (searchQuery) => dispatch(getRecipes(searchQuery))
});

export default connect(null, mapDispatchToProps)(Search);