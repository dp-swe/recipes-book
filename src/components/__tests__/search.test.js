import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Search } from '../Search';

describe('Search Component', () => {
    let submit;
    let wrapper;

    beforeEach(() => {
        submit = jest.fn();
        wrapper = shallow(<Search submit={submit}/>);
    })

    it('should render without broken', () => {
        expect(wrapper.exists()).toBe(true);
    })

    it('should render correctly', () => {
        const tree = renderer.create(<Search submit={submit}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('should not call the submit function when input less than 3 characters', () => {
        wrapper.find('#searchForm').simulate(
            'submit',
            {preventDefault() {}}
        )
        expect(submit.mock.calls.length).toBe(0);
    })

    it('should update state on change of input', () => {
        wrapper.find('#searchInput').simulate('change', {target:{value:'a'}});
        expect(wrapper.find('#searchInput').prop('value')).toEqual('a');
    })

    it('should call the submit function when input more than 3 characters', () => {
        wrapper.find('#searchInput').simulate('change', {target:{value:'abcd'}});
        wrapper.find('#searchForm').simulate(
            'submit',
            {preventDefault() {}}
        )
        expect(submit.mock.calls.length).toBe(1);
    })
})