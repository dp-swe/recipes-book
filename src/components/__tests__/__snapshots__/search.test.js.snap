// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render correctly 1`] = `
<div
  className="search"
>
  <form
    id="searchForm"
    onSubmit={[Function]}
  >
    <div
      className="search__input_parent"
    >
      <svg
        className="search__icon"
      >
        search.svg
      </svg>
      <input
        className="search__input"
        id="searchInput"
        onChange={[Function]}
        value=""
      />
    </div>
  </form>
</div>
`;
